packaging/os/yum.py

```diff
@@ -606,54 +628,56 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
 
         pkgs.append(pkg)
 
-    if pkgs:
-        cmd = yum_basecmd + ['install'] + pkgs
-
-        if module.check_mode:
-            # Remove rpms downloaded for EL5 via url
-            try:
-                shutil.rmtree(tempdir)
-            except Exception, e:
-                module.fail_json(msg="Failure deleting temp directory %s, %s" % (tempdir, e))
+    if not pkgs:
+        delete_temporary_directory(tempdir)
+        return res
 
-            module.exit_json(changed=True, results=res['results'], changes=dict(installed=pkgs))
+    # yum downgrading will not work on lists of packages. This is a necessary
+    # workaround for now since building the logic for this will take a larger
+    # refactoring effort.
+    if downgrade and len(pkgs) == 1:
```

Comment:
What's the anticipated behaviour in this circumstance? Does the yum command fail and pass that back to the module or does it silently continue with no change recorded?